sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -447,7 +388,7 @@ public void processElement(ProcessContext c) {
                         }
                       })
                   .withSideInputs(empty));
-      return materialized.apply(Reshuffle.viaRandomKey());
+      return materialized.apply(Distinct.create());
```

Comment:
why are we using distinct here?

Reply:
I am not ready to mark this as resolved yet - this approach using `Distinct` may be a problem in streaming, so I want to make sure I read the guarantees properly

Reply:
Ok, let's update the documentation of the transform to indicate that redis may output duplicates, and so we have an extra step to deduplicate them. I think it makes sense to add an option `withAllowDuplicates` that increases performance but may have duplicate elements in a single PCollection

Also, instead of `Distinct`, let's use `Latest.perKey()` (https://beam.apache.org/documentation/transforms/java/aggregation/latest/)

Reply:
@chamikaramj what are your thoughts on a parameter `withAllowDuplicates`? Or we could have the fewest possible knobs

Reply:
About using withAllowDuplicates, should we proceed with the change or is there is something else to discuss about it?